mlir-sycl/lib/Dialect/IR/SYCLOpsDialect.cpp

```diff
@@ -125,6 +125,60 @@ void mlir::sycl::SYCLDialect::printType(
   }
 }
 
+llvm::Optional<llvm::StringRef>
+mlir::sycl::SYCLDialect::findMethod(mlir::TypeID BaseType,
+                                    llvm::StringRef MethodName) const {
+  return syclCtx.findMethod(BaseType, MethodName);
+}
+
+llvm::Optional<llvm::StringRef>
+mlir::sycl::SYCLContext::findMethod(mlir::TypeID BaseType,
+                                    llvm::StringRef MethodName) const {
+  return methodRegistry.lookupMethod(BaseType, MethodName);
+}
+
+llvm::Optional<llvm::StringRef>
+mlir::sycl::MethodRegistry::lookupMethod(mlir::TypeID BaseType,
+                                         llvm::StringRef MethodName) const {
+  const auto Iter = methods.find({BaseType, MethodName});
+  return Iter == methods.end() ? llvm::None
+                               : llvm::Optional<llvm::StringRef>{Iter->second};
+}
+
+void mlir::sycl::SYCLContext::addMethod(mlir::TypeID TypeID,
+                                        llvm::StringRef MethodName,
+                                        llvm::StringRef OpName) {
+  methodRegistry.registerMethod(TypeID, MethodName, OpName);
+}
+
+void mlir::sycl::MethodRegistry::registerMethod(mlir::TypeID TypeID,
```

Comment:
How about making this member function return a bool (true is the method was registered and false if registration fails). The caller can then check the returned value and bail out with an error if the method registration failed for any reason.

Reply:
Good idea